airflow/providers/amazon/aws/sensors/utils.py

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+
+
+def raise_failed_or_skiping_exception(
+    *, soft_fail: bool = False, failed_message: str, skipping_message: str = ""
+) -> None:
+    """Raise AirflowSkipException if self.soft_fail is set to True. Otherwise raise AirflowException."""
+    if soft_fail:
+        raise AirflowSkipException(skipping_message or failed_message)
+    raise AirflowException(failed_message)
```

Comment:
so basically what you are after is to backport a fix we have in 2.7.1
The proper way to do it is:

1. Try to import the function from base_sensor if failed on import error then use the local version.
2. The local version should be private `_raise_failed_or_skiping_exception` so we can remove it without deprecation cycle.
3. There should be a note on the function to remove it when min airflow version for provider is >=2.7.1

Reply:
> Try to import the function from base_sensor if failed on import error then use the local version.

This method was removed from base sensor after merging #33424.

> The local version should be private _raise_failed_or_skiping_exception so we can remove it without deprecation cycle.
There should be a note on the function to remove it when min airflow version for provider is >=2.7.

a better solution could be removing this method and duplicating this small block in all the operators, it's the most safe solution to avoid breaking changes. WDYT?

Reply:
I didn't track the flow of PR around the soft_fail but that is exactly what troubles me.
 I can be the one that pick up the todo in future releases when we want to do the cleanup. I do not understand what happens here. What is the old and what is the new.
 
 I trust you guys have clarity on this thus i'm removing my review and you can proceed as you see fit (as I'm not in the details)

Reply:
Basically we fixed the issue in a more proper way in base sensor, and all the providers will work fine with Airflow +2.7.1, but to fix this issue with older versions of Airflow, we need to implement a temporary solution in each one until bumping the min Airflow version for these providers to 2.7.1